force-app/main/default/classes/AttendanceService_TEST.cls

```diff
@@ -324,6 +325,32 @@ public with sharing class AttendanceService_TEST {
         );
     }
 
+    @IsTest
+    private static void shouldCallPermissionValidator() {
+        String methodName = 'hasFLSAccessForFields';
+        permValidatorStub.withReturnValue(
+            methodName,
+            new List<Type>{ List<Schema.SObjectField>.class, String.class },
+            true
+        );
+        PermissionValidator.instance = (PermissionValidator) permValidatorStub.createMock();
+
+        Map<String, Boolean> expected = new Map<String, Boolean>{
+            'read' => true,
+            'write' => true
+        };
+
+        Test.startTest();
+        Map<String, Boolean> actual = service.checkFieldPermissions();
+        Test.stopTest();
+
+        permValidatorStub.assertCalled(
+            methodName,
+            new List<Type>{ List<Schema.SObjectField>.class, String.class }
+        );
+        System.assertEquals(expected, actual);
```

Comment:
I think the results of the expected vs actual will give us enough information if this where to fail but I know we also talked about adding a message to all asserts. I will leave it up to you.